test/spec/modules/conversantAnalyticsAdapter_spec.js

```diff
@@ -0,0 +1,733 @@
+import * as caObj from 'modules/conversantAnalyticsAdapter.js';
+import * as utils from 'src/utils.js';
+import * as prebidGlobal from 'src/prebidGlobal';
+import events from 'src/events';
+import constants from 'src/constants.json'
+
+describe('Conversant analytics adapter tests', function() {
+  let sandbox; // sinon sandbox to make restoring all stubbed objects eaiser
+  let xhr; // xhr stub from sinon for capturing data sent via ajax
+  let clock; // clock stub from sinon to mock our cache cleanup interval
+
+  const PREBID_VERSION = '1.2';
+  const SITE_ID = 108060;
+
+  let requests = [];
+  const DATESTAMP = Date.now();
+
+  let VALID_CONFIGURATION = {
+    options: {
+      site_id: SITE_ID
+    }
+  };
+
+  let VALID_ALWAYS_SAMPLE_CONFIG = {
+    options: {
+      site_id: SITE_ID,
+      sampleRate: 100
+    }
+  };
+
+  beforeEach(function () {
+    sandbox = sinon.sandbox.create();
+    sandbox.stub(events, 'getEvents').returns([]); // need to stub this otherwise unwanted events seem to get fired during testing
+    xhr = sandbox.useFakeXMLHttpRequest(); // allows us to capture ajax requests
+    xhr.onCreate = function (req) { requests.push(req); }; // save ajax requests in a private array for testing purposes
+    let getGlobalStub = {
+      version: PREBID_VERSION,
+      getUserIds: function() { // userIdTargeting.js init() gets called on AUCTION_END so we need to mock this function.
+        return {};
+      }
+    };
+    sandbox.stub(prebidGlobal, 'getGlobal').returns(getGlobalStub); // getGlobal does not seem to be available in testing so need to mock it
+    clock = sandbox.useFakeTimers(DATESTAMP); // to use sinon fake timers they MUST be created before the interval/timeout is created in the code you are testing.
+  });
+
+  afterEach(function () {
+    sandbox.restore();
+    requests = []; // clean up any requests in our ajax request capture array.
+  });
+
+  describe('Initialization Tests', function() {
+    it('should log error if publisher id is not passed', function() {
+      sandbox.stub(utils, 'logError');
+
+      conversantAnalytics.enableAnalytics();
+      expect(utils.logError.calledWith(caObj.LOG_PREFIX + 'siteId is required.')).to.be.true;
+      conversantAnalytics.disableAnalytics();
+    });
+
+    it('should not log error if valid config is passed', function() {
+      sandbox.stub(utils, 'logError');
+      sandbox.stub(utils, 'logInfo');
+
+      conversantAnalytics.enableAnalytics(VALID_CONFIGURATION);
+      expect(utils.logError.called).to.equal(false);
+      expect(utils.logInfo.called).to.equal(true);
+      expect(
+        utils.logInfo.calledWith(
+          caObj.LOG_PREFIX + 'Sample rate set to ' + caObj.DEFAULT_SAMPLE_RATE + '%'
+        )
+      ).to.be.true;
+      conversantAnalytics.disableAnalytics();
+    });
+
+    it('should sample when sampling set to 1', function() {
+      sandbox.stub(utils, 'logError');
+
+      VALID_CONFIGURATION.options.sampleRate = 100;
```

Comment:
Probably best to make a deep clone of VALID_CONFIGURATION and modify the copy instead.  Use utils.deepClone.